File: testClient.py
#! /usr/bin/env python
#encoding: utf-8

import sys
import numpy as np
import cv2
from remoteFrame import *

port = 8888
address = '127.0.0.1'
Mask = None

argc = len (sys.argv)
if argc > 1:
    address = sys.argv[1]
    if argc > 2:
        port = int(sys.argv[2])
        if argc > 3:
            Mask = cv2.imread(sys.argv[3])

rF=remoteFrame(address, port, Mask)

cv2.namedWindow('MyWindow')
img =cv2.Mat()
while cv2.waitKey(1) == -1: 
    img=rF.getFrame()
    cv2.imshow('MyWindow', img)

File: testClient.cpp
/*!
\file testClient.h
\brief Este es un programa de ejemplo en donde se muestra como un programa puede recibir imágenes del servidor y mostrarlas en una ventana.
*/

#include <Client.h>
#include <opencv2/core/core.hpp>
#include <opencv2/imgproc/imgproc.hpp>
#include <opencv2/highgui/highgui.hpp>

int main (int argc, char **argv)
{
	 int port = 8888;
	 char ipAddress[64] = "127.0.0.1";

	 if (argc > 1)
	 {
			strncpy (ipAddress, argv[1], 63);
			if (argc > 2)
				 port = atoi (argv[2]);
	 }

	 Client *client = new Client (port, ipAddress);
	 client->connectSocket ();

	 Mat frame;
	 while (true)
	 {
			client->getFrame (frame);
			imshow ("rec", frame);
			if (waitKey (30) >= 0)
				 break;
	 }


	 return 0;
}
